### PR TITLE
Better database errors

### PR DESCRIPTION
**Pull request by cmbrose**
## What

Extended the postgres package to capture pq Errors and turn them into database errors
## Why

Allows database users to not have to care about what driver is being used for error checking.  All errors should either be SQL or database types.
## How

postgres catches errors and checks if it is a pq.Error.  If it is, it simply swaps out that error for a better (more user friendly) one.

### CODE REVIEW

**Comment by cmbrose** [comment]
@ngmiller this should fix your duplicate alias error code issue

File: databases/postgres/postgres.go [hunk 1]
```diff
@@ -21,21 +21,55 @@ import (
...
+        pqErr, ok = err.(*pq.Error)
+    } 
+
+    if !ok {
+        return nil
+    }
+
+    switch pqErr.Code {
+    case "23505": 
+        return databases.DuplicateKeyError
```
Comment (cmbrose) [inline databases/postgres/postgres.go:66]: @ngmiller @zach-taylor I'm only catching this one specific error currently, so if you've seen other error text that looks like `pq: ...`, let me know and I'll add it here.

**Comment by cmbrose** [comment]
@ngmiller @zach-taylor were there any other errors you've seen coming up that should get patched?  Otherwise I'd like to get this guy through.

**Comment by nickmonad** [comment]
Nope I think that's the only one I've seen thus far. Can always add more later. +1

**Comment by chfogerty** [comment]
+1

**Pull request merged**